### ISSUE TITLE
TorrentToMedia not processing single-file downloads?

### ISSUE THREAD

**Issue by Mini-Myself** [clinton-hall/nzbToMedia#1796]
**Describe the bug**
I have been using the script for a week now, tried several torrent clients (utorrent, qbittorrent, deluge).
All seem to be having the same issue, so I must be doing SOMETHING wrong but I can't figure it out.

When a torrent is only 1 file (no folder) it does not seem to be triggering the hard-link/copy/move (does not matter what option I set in the config).
Because of this, Medusa finds an empty folder and does not process anything.

Config file:
[SickBeard]
    [[tv]]
        enabled = 1
        fork = medusa
        delete_failed = 0
        Torrent_NoLink = 0
        keep_archive = 1
        process_method = ""
        delete_on = 1
        extract = 1
        watch_dir = C:\Users\x\Downloads\tv
[Torrent]
    clientAgent = deluge
    useLink = hard
    outputDirectory = C:\Users\x\Downloads
    default_downloadDirectory = C:\Users\x\Downloads

**Technical Specs**
1. Running on Windows 10
2. Python version 3.7.9
3. Download Client Deluge now, QBittorrent before, µtorrent before that
4. Intended Media Management Medusa

**Expected behavior**
I download with the category "tv", so downloads are place in "C:\Users\x\Downloads\tv". I want to keep seeding so a copy/hard-link is needed. The script seems to default to "C:\Users\x\Downloads\tv\tv" which is fine. The files need to be hard-linked here so Medusa can be triggered to move the files to my HDD which is located at "D:\Series\*".

**Log**
2020-11-15 13:51:44 INFO    ::MAIN: #########################################################
2020-11-15 13:51:44 INFO    ::MAIN: ## ..::[TorrentToMedia.pyw]::.. ##
2020-11-15 13:51:44 INFO    ::MAIN: #########################################################
2020-11-15 13:51:44 DEBUG   ::MAIN: Options passed into TorrentToMedia: ['C:\\Medusa\\Scripts\\nzbToMedia\\TorrentToMedia.pyw', '3b871bd161765bc99db7c7834d76862d3489b43c', '[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv', 'C:\\Users\\x\\Downloads\\tv']
2020-11-15 13:51:44 DEBUG   ::MAIN: Adding TORRENT download info for directory C:\Users\x\Downloads\tv to database
2020-11-15 13:51:44 DB      ::MAIN: nzbtomedia.db: UPDATE downloads SET input_name = ?, input_hash = ?, input_id = ?, client_agent = ?, status = ?, last_update = ? WHERE input_directory = ? with args ['[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv', '3b871bd161765bc99db7c7834d76862d3489b43c', '3b871bd161765bc99db7c7834d76862d3489b43c', 'deluge', 0, 737744, 'C:\\Users\\x\\Downloads\\tv']
2020-11-15 13:51:44 DB      ::MAIN: nzbtomedia.db: INSERT OR IGNORE INTO downloads (input_name, input_hash, input_id, client_agent, status, last_update) VALUES (?, ?, ?, ?, ?, ?) with args ['[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv', '3b871bd161765bc99db7c7834d76862d3489b43c', '3b871bd161765bc99db7c7834d76862d3489b43c', 'deluge', 0, 737744]
2020-11-15 13:51:44 DEBUG   ::MAIN: Received Directory: C:\Users\x\Downloads\tv | Name: [Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv | Category: 
2020-11-15 13:51:44 DEBUG   ::MAIN: SEARCH: Found Category: tv in directory structure
2020-11-15 13:51:44 INFO    ::MAIN: SEARCH: Found category directory tv in input directory directory C:\Users\x\Downloads\tv
2020-11-15 13:51:44 INFO    ::MAIN: SEARCH: Setting input_directory to C:\Users\x\Downloads\tv\tv
2020-11-15 13:51:44 INFO    ::MAIN: SEARCH: Could not find a unique directory for this download. Assume a common directory.
2020-11-15 13:51:44 INFO    ::MAIN: SEARCH: We will try and determine which files to process, individually
2020-11-15 13:51:44 DEBUG   ::MAIN: Determined Directory: C:\Users\x\Downloads\tv\tv | Name: [Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv | Category: tv
2020-11-15 13:51:44 INFO    ::MAIN: Auto-detected SECTION:SickBeard
2020-11-15 13:51:44 DEBUG   ::MAIN: Stopping torrent [Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv in deluge while processing
2020-11-15 13:51:44 WARNING ::MAIN: Failed to stop torrent [Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv in deluge
2020-11-15 13:51:44 INFO    ::MAIN: Output directory set to: C:\Users\x\Downloads\tv\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle]
2020-11-15 13:51:44 DEBUG   ::MAIN: Scanning files in directory: C:\Users\x\Downloads\tv\tv
2020-11-15 13:51:44 DEBUG   ::MAIN: Found 0 files in C:\Users\x\Downloads\tv\tv
2020-11-15 13:51:44 DEBUG   ::MAIN: Checking for archives to extract in directory: C:\Users\x\Downloads\tv\tv
2020-11-15 13:51:44 INFO    ::MAIN: FLATTEN: Flattening directory: C:\Users\x\Downloads\tv\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle]
2020-11-15 13:51:44 WARNING ::MAIN: Found no media files in C:\Users\x\Downloads\tv\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle]
2020-11-15 13:51:44 INFO    ::MAIN: Calling SickBeard:tv to post-process:[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv
2020-11-15 13:51:44 DEBUG   ::SERVER: Attempting to connect to server at https://localhost:8081
2020-11-15 13:51:46 DEBUG   ::SERVER: Server responded at https://localhost:8081
2020-11-15 13:51:46 INFO    ::MAIN: SickBeard:tv fork set to Medusa
2020-11-15 13:51:46 WARNING ::SICKBEARD: No media files found in directory C:\Users\x\Downloads\tv\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle]. Processing this as a failed download
2020-11-15 13:51:46 POSTPROCESS::SICKBEARD: FAILED: The download failed. Sending 'failed' process request to Medusa branch
2020-11-15 13:51:46 DEBUG   ::SICKBEARD: Opening URL: https://localhost:8081/home/postprocess/processEpisode with params: {'proc_dir': 'C:\\Users\\x\\Downloads\\tv\\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle]', 'failed': 1, 'force': 1, 'delete_on': 1, 'quiet': 1, 'nzbName': '[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv'}
2020-11-15 13:51:49 POSTPROCESS::SICKBEARD: Processing path: C:\Users\x\Downloads\tv\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle]
2020-11-15 13:51:49 POSTPROCESS::SICKBEARD: Post-processing completed.
2020-11-15 13:51:49 POSTPROCESS::SICKBEARD: I did encounter some unprocessable items: 
2020-11-15 13:51:49 POSTPROCESS::SICKBEARD: Missed file: C:\Users\x\Downloads\tv\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle]: Failed download
2020-11-15 13:51:49 ERROR   ::MAIN: A problem was reported in the autoProcess* script. Torrent won't resume seeding (settings)
2020-11-15 13:51:49 ERROR   ::MAIN: A problem was reported in the C:\Medusa\Scripts\nzbToMedia\TorrentToMedia.pyw script.

**Comment by clinton-hall** [comment]
Ok, having \tv\tv really messes with the script. So remove the extra "tv" directory, it should be empty.
To prevent this from being re-created, set `outputDirectory = C:\Users\x\Processed` so that way the media will be hard-linked into `C:\Users\x\Processed\tv\<download name>\<file name>.ext` and then sent to Medusa.

Also, I note it fails to stop the torrent... Failed to stop torrent [Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv in deluge
So check the `[Torrent]` settings to make sure you have the correct deluge details
```
[Torrent]
    DelugeHost = localhost
    DelugePort = 58846
    DelugeUSR = your username
    DelugePWD = your password
```
and make sure your deluge `%APPDATA%\deluge\auth` file has the entry `<username>:<password>:10` to allow access to deluge for this user/pass.

Once you have made those changes, try again and post nzbtomedia.log for another run if this still doesn't work.

**Comment by Mini-Myself** [comment]
When there is no folder called TV it does not detect the category and derps out. I fixed the connection with Deluge by the way (but that is not visible in this log now, hehe)

```
2020-11-15 22:50:19 INFO    ::MAIN: #########################################################
2020-11-15 22:50:19 INFO    ::MAIN: ## ..::[TorrentToMedia.pyw]::.. ##
2020-11-15 22:50:19 INFO    ::MAIN: #########################################################
2020-11-15 22:50:19 DEBUG   ::MAIN: Options passed into TorrentToMedia: ['C:\\Medusa\\Scripts\\nzbToMedia\\TorrentToMedia.pyw', '3b871bd161765bc99db7c7834d76862d3489b43c', '[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv', 'C:\\Users\\x\\Downloads\\seeding']
2020-11-15 22:50:19 DEBUG   ::MAIN: Adding TORRENT download info for directory C:\Users\x\Downloads\seeding to database
2020-11-15 22:50:19 DB      ::MAIN: nzbtomedia.db: UPDATE downloads SET input_name = ?, input_hash = ?, input_id = ?, client_agent = ?, status = ?, last_update = ? WHERE input_directory = ? with args ['[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv', '3b871bd161765bc99db7c7834d76862d3489b43c', '3b871bd161765bc99db7c7834d76862d3489b43c', 'deluge', 0, 737744, 'C:\\Users\\x\\Downloads\\seeding']
2020-11-15 22:50:19 DB      ::MAIN: nzbtomedia.db: INSERT OR IGNORE INTO downloads (input_name, input_hash, input_id, client_agent, status, last_update) VALUES (?, ?, ?, ?, ?, ?) with args ['[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv', '3b871bd161765bc99db7c7834d76862d3489b43c', '3b871bd161765bc99db7c7834d76862d3489b43c', 'deluge', 0, 737744]
2020-11-15 22:50:19 DEBUG   ::MAIN: Received Directory: C:\Users\x\Downloads\seeding | Name: [Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv | Category: 
2020-11-15 22:50:19 DEBUG   ::MAIN: SEARCH: Could not find a category in the directory structure
2020-11-15 22:50:19 INFO    ::MAIN: SEARCH: Found torrent file [Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv in input directory directory C:\Users\x\Downloads\seeding
2020-11-15 22:50:19 INFO    ::MAIN: SEARCH: Setting input_directory to C:\Users\x\Downloads\seeding\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv
2020-11-15 22:50:19 DEBUG   ::MAIN: Determined Directory: C:\Users\x\Downloads\seeding\[Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv | Name: [Erai-raws] One Piece - 950 [1080p][Multiple Subtitle].mkv | Category: UNCAT
2020-11-15 22:50:19 ERROR   ::MAIN: Unable to locate a section with subsection:UNCAT enabled in your autoProcessMedia.cfg, exiting!

**Comment by Mini-Myself** [comment]
Okay so I managed to get a working situation.
However I had to switch to qBittorrent since that receives the TV category from the script rather than from the directory structure. 
Also I had to move "C:\Users\x\Downloads\seeding\tv" to "C:\Users\x\Downloads\tv". Because otherwise the script would find the folder sitting there and not do any hard-linking.

I can still run tests if you like with Deluge (I will just make a copy of my config). However the issue is resolved.
